### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by Kirovtls**

### CODE REVIEW

File: src/main/java/kg/peaksoft/taskTrackerb6/db/service/AdminService.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        return projectResponses;
+    }
+
+    private ProjectResponse convertToProjectResponse(Workspace workspace) {
+        return new ProjectResponse(workspace.getName());
+    }
+
+
+    public AdminProfileResponse changePhoto(Long id,String photo) {
```
Comment (DatkaM) [inline src/main/java/kg/peaksoft/taskTrackerb6/db/service/AdminService.java:61]: remove method

File: src/main/java/kg/peaksoft/taskTrackerb6/dto/request/AdminProfileRequest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package kg.peaksoft.taskTrackerb6.dto.request;
+
+import lombok.Getter;
+import lombok.Setter;
+
+
+@Getter
+@Setter
+public class AdminProfileRequest {
```
Comment (DatkaM) [inline src/main/java/kg/peaksoft/taskTrackerb6/dto/request/AdminProfileRequest.java:9]: add field for photo

File: src/main/java/kg/peaksoft/taskTrackerb6/db/service/AdminService.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        user1.setPhotoLink(photo);
+        return new AdminProfileResponse(
+                user1.getId(),
+                user1.getFirstName(),
+                user1.getLastName(),
+                user1.getEmail(),
+                user1.getPhotoLink(),
+                getAllProjectResponse());
+    }
+
```
Comment (DatkaM) [inline src/main/java/kg/peaksoft/taskTrackerb6/db/service/AdminService.java:80]: remove method

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import io.swagger.v3.oas.annotations.tags.Tag;
+import kg.peaksoft.taskTrackerb6.db.service.AdminService;
+import kg.peaksoft.taskTrackerb6.dto.request.AdminProfileRequest;
+import kg.peaksoft.taskTrackerb6.dto.response.AdminProfileResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/admin/profile")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:13]: Write api/profile only

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Admin profile Api", description = "All endpoints of admin profile")
+public class AdminProfileApi {
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:16]: ProfileApi only

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import kg.peaksoft.taskTrackerb6.dto.request.AdminProfileRequest;
+import kg.peaksoft.taskTrackerb6.dto.response.AdminProfileResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/admin/profile")
+@CrossOrigin(origins = "*", maxAge = 3600)
+@Tag(name = "Admin profile Api", description = "All endpoints of admin profile")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:15]: Profile API

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    private final AdminService adminService;
+
+    @Operation(summary = "Get profile", description = "Get profile by id")
+    @GetMapping("{id}")
+    public AdminProfileResponse getProfile(@PathVariable Long id) {
+        return adminService.adminProfile(id);
+    }
+
+    @Operation(summary = "Update profile", description = "Update profile by profile id")
+    @PutMapping("update")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:27]: Endpoint naming convention

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    public AdminProfileResponse updateAdminProfile(@RequestBody AdminProfileRequest request) {
+        return adminService.updateUserEntity(request);
+    }
+
+    @Operation(summary = "Change photo", description = "Change profile photo by profile id")
+    @PutMapping("change-photo/{id}")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:33]: Endpoint naming convention

File: src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    public AdminProfileResponse changeProfilePhoto(@PathVariable Long id,
+                                                   @RequestBody String photo) {
+        return adminService.changePhoto(id, photo);
+    }
+
+    @Operation(summary = "Delete photo", description = "Delete profile photo by profile id")
+    @PutMapping("remove-photo/{id}")
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/taskTrackerb6/api/AdminProfileApi.java:40]: @PutMapping?

**Review by ChyngyzIshen — COMMENTED** [review]
Remove changePhoto removePhoto methods

**Pull request merged**